app/main.py

```diff
@@ -1,25 +1,96 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: SlideLimitationValidator,
+            name: str
+    ) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: SlideLimitationValidator
+    ) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(
+            self,
+            instance: SlideLimitationValidator,
+            value: int
+    ) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
-class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+class ChildrenSlideLimitationValidator(
+    SlideLimitationValidator
+):
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        super().__init__(age, weight, height)
 
-class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+
+class AdultSlideLimitationValidator(
+    SlideLimitationValidator
+):
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
+
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        super().__init__(age, weight, height)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: SlideLimitationValidator
+    ) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> Visitor | None:
+        result = (self.limitation_class(
+            visitor.age,
+            visitor.weight,
+            visitor.height
+        ))
+        if len(result.__dict__) == 3:
```

Comment:
This is a code quality issue. Checking the length of the `__dict__` attribute to determine if the object is valid is not a good practice. It might be better to define a method in the `SlideLimitationValidator` class to check if the object is valid, and call that method here.